processor/src/main/java/org/mapstruct/ap/internal/conversion/BigIntegerToStringConversion.java

```diff
@@ -57,22 +60,21 @@ public String getToExpression(ConversionContext conversionContext) {
     public String getFromExpression(ConversionContext conversionContext) {
         if ( requiresDecimalFormat( conversionContext ) ) {
             StringBuilder sb = new StringBuilder();
-            sb.append( "( (BigDecimal) " );
+            sb.append( "( (" + bigDecimal( conversionContext ) + ") " );
             appendDecimalFormatter( sb, conversionContext );
             sb.append( ".parse( <SOURCE> )" );
             sb.append( " ).toBigInteger()" );
             return sb.toString();
         }
         else {
-            return "new BigInteger( <SOURCE> )";
+            return "new " + bigInteger( conversionContext ) + "( <SOURCE> )";
         }
     }
 
     @Override
     protected Set<Type> getFromConversionImportTypes(ConversionContext conversionContext) {
         if ( requiresDecimalFormat( conversionContext ) ) {
-            // no imports are required when decimal format is used.
-            return super.getFromConversionImportTypes( conversionContext );
```

Comment:
Shouldn't we get the one from the parent too?

Reply:
Hmm, you mean that `super.getFromConversionImportTypes` should be called?
I don't think this is required as `getFromExpression` isn't using the `DecimalFormat` that would be returned from `super.getFromConversionImportTypes`. The `appendDecimalFormatter` method is adding a call to a helper method that has it's own import types.

Reply:
And again you are right 😄. Didn't analyze it that deep 😄

Reply:
But it was definitly a good point. I also had to go through the code again and was wondering why I removed the `super` call.